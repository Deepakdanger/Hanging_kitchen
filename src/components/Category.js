import React from 'react';
import PropTypes from 'prop-types';

const Category = ({ ele, selectCategory }) => {
  const { strCategoryThumb, strCategory } = ele;
  return (
    <div className="category_list swing">
      <img className="category_img" src={strCategoryThumb} alt="NA" />
      <button className="category_name" type="button" onClick={selectCategory}>{strCategory}</button>
    </div>
  );
};

Category.propTypes = {
  ele: PropTypes.shape({
    strCategoryThumb: PropTypes.string,
    strCategory: PropTypes.string,
  }),
  selectCategory: PropTypes.func.isRequired,
};

Category.defaultProps = {
  ele: PropTypes.shape({
    strCategoryThumb: '1',
    strCategory: 'fhfh',
  }),
};

export default Category;
