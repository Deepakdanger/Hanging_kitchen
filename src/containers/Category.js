import React from 'react';
import PropTypes from 'prop-types';

const Catagory = ({ ele }) => {
  const { idCategory, strCategory } = ele;
  return (
    <div>
      <span>{idCategory}</span>
      <span>{strCategory}</span>
    </div>
  );
};
Catagory.propTypes = {
  ele: PropTypes.shape({
    idCategory: PropTypes.string.isRequired,
    strCategory: PropTypes.string.isRequired,
  }),
};

export default Catagory;
