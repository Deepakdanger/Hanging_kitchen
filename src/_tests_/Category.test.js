import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import Category from '../components/Category';

describe('Testing Filter with static router', () => {
  it('Category component is defined', async () => {
    const component = render(
      <StaticRouter>
        <Category ele={{}} selectCategory={jest.fn()} />
      </StaticRouter>,
    );
    expect(component).toBeDefined();
  });

  it('Filter snapshot testing', () => {
    const tree = renderer.create(
      <StaticRouter>
        <Category ele={{}} selectCategory={jest.fn()} />
      </StaticRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
