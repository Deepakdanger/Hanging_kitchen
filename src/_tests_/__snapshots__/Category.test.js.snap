// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filter with static router Category snapshot testing 1`] = `
<div
  className="category_list swing"
>
  <img
    alt="NA"
    className="category_img"
  />
  <button
    className="category_name"
    onClick={[MockFunction]}
    type="button"
  />
</div>
`;

exports[`Testing Filter with static router Filter snapshot testing 1`] = `
<div
  className="category_list swing"
>
  <img
    alt="NA"
    className="category_img"
  />
  <button
    className="category_name"
    onClick={[MockFunction]}
    type="button"
  />
</div>
`;
